fix: deploy-validation CI missing bun install for create-atlas deps

## Problem

The new deploy-validation workflow (PR #1114) fails on the scaffold smoke test because `@clack/prompts` is not installed. The workflow runs `create-atlas/smoke-test.sh` which invokes `bun run create-atlas/index.ts`, but only the workspace-level `bun install` is run — create-atlas has its own dependencies that need resolving.

## Error

```
error: Cannot find module '@clack/prompts' from '/home/runner/work/atlas/atlas/create-atlas/index.ts'
```

## Fix

Add `cd create-atlas && bun install && cd ..` before the scaffold smoke test step, or run `bun install --filter create-atlas` in the setup step.

## Impact

New Deploy Validation workflow always fails. Main CI is unaffected.

--- a/.github/workflows/deploy-validation.yml
+++ b/.github/workflows/deploy-validation.yml
@@ -53,7 +53,7 @@ jobs:
         run: bun install --frozen-lockfile
 
       - name: Install create-atlas dependencies
-        run: cd create-atlas && bun install
+        run: cd create-atlas && bun install --frozen-lockfile
 
       - name: Run scaffold smoke test
         run: bash create-atlas/smoke-test.sh ${{ matrix.platform }}

--- a/packages/api/src/lib/__tests__/settings.test.ts
+++ b/packages/api/src/lib/__tests__/settings.test.ts
@@ -276,6 +276,27 @@ describe("settings module", () => {
       mockPool.query = savedQuery;
     });
 
+    it("atomic swap — error during reload preserves old cache", async () => {
+      enableInternalDB();
+
+      // Load initial data
+      setResults({
+        rows: [
+          { key: "ATLAS_ROW_LIMIT", value: "100", updated_at: "2026-01-01", updated_by: null, org_id: null },
+        ],
+      });
+      await loadSettings();
+      expect(getSetting("ATLAS_ROW_LIMIT")).toBe("100");
+
+      // Next load throws
+      queryThrow = new Error("connection reset by peer");
+      const count = await loadSettings();
+      expect(count).toBe(0);
+
+      // Old cache value is still readable (not wiped)
+      expect(getSetting("ATLAS_ROW_LIMIT")).toBe("100");
+    });
+
     it("atomic swap — stale entries are removed (full replacement, not merge)", async () => {
       enableInternalDB();
 

--- a/packages/api/src/lib/settings.ts
+++ b/packages/api/src/lib/settings.ts
@@ -324,6 +324,12 @@ const SETTINGS_REGISTRY: SettingDefinition[] = [
 // In-process cache
 // ---------------------------------------------------------------------------
 
+interface CacheEntry {
+  value: string;
+  updated_at: string;
+  updated_by: string | null;
+}
+
 /**
  * Cache key format:
  * - Platform (global): "KEY"
@@ -334,7 +340,7 @@ function cacheKey(key: string, orgId?: string | null): string {
   return orgId ? `${key}\0${orgId}` : key;
 }
 
-let _cache = new Map<string, { value: string; updated_at: string; updated_by: string | null }>();
+let _cache = new Map<string, CacheEntry>();
 
 const SETTINGS_MAP = new Map(SETTINGS_REGISTRY.map((s) => [s.key, s]));
 
@@ -440,7 +446,7 @@ export async function loadSettings(): Promise<number> {
       "SELECT key, value, updated_at::text, updated_by, org_id FROM settings",
     );
 
-    const next = new Map<string, { value: string; updated_at: string; updated_by: string | null }>();
+    const next = new Map<string, CacheEntry>();
     for (const row of rows) {
       next.set(cacheKey(row.key, row.org_id), {
         value: row.value,
@@ -455,6 +461,7 @@ export async function loadSettings(): Promise<number> {
     }
     return rows.length;
   } catch (err) {
+    // On error, _cache is unchanged — atomic swap ensures readers see last successful load
     const msg = err instanceof Error ? err.message : String(err);
     // "42P01" = relation does not exist — expected on first boot before migration
     const isTableMissing = msg.includes("does not exist") || msg.includes("42P01");

